stream_alert/rule_processor/rules_engine.py

```diff
@@ -443,14 +439,20 @@ def rule_analysis(record, rule, payload, alerts):
             LOGGER.info('Rule [%s] triggered alert [%s] on log type [%s] from entity \'%s\' '
                         'in service \'%s\'', rule.rule_name, alert_id, payload.log_source,
                         payload.entity, payload.service())
+
+            # Combine the required alert outputs with the ones for this rule
+            all_outputs = self._required_outputs
+            if rule.outputs:
+                all_outputs = all_outputs.union(set(rule.outputs))
```

Comment:
When I saw this, I thought it was a bug because it looks like modifying `all_outputs` should modify `self._required_outputs` (they are initially the same reference). But this line actually rebinds `all_outputs` to reference a new set created by the union.

For clarity, and to avoid possible future bugs with this, can we just do
```python
all_outputs = self._required_outputs | set(rule.outputs)
```

Ideally, `rule.outputs` is always a list or set, and that will work fine. But if not, you can do
```python
all_outputs = self._required_outputs | set(rule.outputs or [])
```

Reply:
ah thank you! I was actually trying to figure out the best way to ensure rule outputs wasn't None